snooty/types.py

```diff
@@ -66,10 +66,33 @@ def without_known_suffix(self) -> str:
         return fileid.as_posix()
 
 
-@dataclass
 class Diagnostic:
-    __slots__ = ("message", "severity", "start", "end")
+    # instance properties
+    __slots__ = ("message", "start", "end")
```

Comment:
Just drop the __slots__ attribute, and the preceding comment. It's a space optimization that isn't actually necessary here, and won't really work anymore anyway with subclasses unless you add __slots__ to all of them as well